### DISCUSSION TITLE
Windows Defender identifying Mirth Connect Administrator Launcher v1.4.1 as being vulnerable under CVE-2023-43208

### DISCUSSION THREAD

**Discussion by pduncliffe** [nextgenhealthcare/connect#6216] [Q&A]
It appears Windows Defender is falsely identifying Mirth Connect Administrator Launcher v1.4.1 as being vulnerable under CVE-2023-43208. Has anyone else dealt with this or reported to Microsoft? It appears to have started after [https://learn.microsoft.com/en-us/defender-vulnerability-management/fixed-reported-inaccuracies] was posted.
![image](https://github.com/nextgenhealthcare/connect/assets/171677948/54190d83-8545-4e2b-83ec-853c7a122e2e)

**Comment by rjkroll** [comment]
It is probably showing up now because this was just discovered last week it seems:

https://nvd.nist.gov/vuln/detail/CVE-2023-43208

"NextGen Healthcare Mirth Connect before version 4.4.1 is vulnerable to unauthenticated remote code execution. Note that this vulnerability is caused by the incomplete patch of CVE-2023-37679."

**Reply by pduncliffe** [reply]
The issue is shown below. Defender appears to highlight the launcher as being vulnerable (which as far as I am aware it is not). Is this because the underlying Mirth Connect Launcher package identifies itself as Mirth Connect? Or is this an issue with Defender not distinguishing between the two applications?
![image](https://github.com/nextgenhealthcare/connect/assets/171677948/374ce4f4-2b65-49ed-8004-18d9a527180b)

**Reply by jonbartels** [reply]
> the launcher as being vulnerable

Does the launcher have files from an affected version in its local cache?

**Reply by DanEdomHodge** [reply]
@pduncliffe did you find a resolution for this? We're seeing the same problem. At this point our choices are to uninstall the Administrator launcher or shut down the server.

**Reply by pduncliffe** [reply]
No, no resolution. No comments from NextGen either, nor MS.

**Reply by jonbartels** [reply]
> Does the launcher have files from an affected version in its local cache?

@pduncliffe and @DanEdomHodge - Before you get all gloomy, can you get your scan tools to say exactly which file(s) are being detected for the CVE within the Launcher?

When the Launcher starts the actual Admin client, it downloads and caches JARs for the Admin client. That cache is checksummed and has signatures validated. Then on subsequent launches, the cached files are used for a faster start. If an older version of MC was used at some point in the past, then there may still be old files in cache that are tripping your security scanners.

If it is detecting cached files, then just nuke the cache directory or the targeted files.

If your security scanner won't tell you what files are suspect then try:
* just nuking the cache anyway
* uninstalling and reinstalling MCAL, run the scan on a clean install BEFORE you launch anything with it. If MCAL has never been used to launch the Admin then it has to be caching. If MCAL does trip the scan, then MCAL itself includes these files.

> At this point our choices are to uninstall the Administrator launcher

If you think the problem is MCAL (the launcher), try the alternative Ballista launcher from https://github.com/kayyagari/ballista . It's faster and has a few other options. It is also fully open sourced unlike MCAL. Also since its written in Rust, Ballista itself won't trigger Java CVEs for anything other than cached files. MCAL is also Java so it can be confusing to sort out if a vuln is from cache or from the MCAL app itself.

> No comments from NextGen either, nor MS.

Can you share references or links to where you have shared this with NG and MS? (Assuming they are like public forums and not private tickets)

**Reply by JackieK5** [reply]
We have reviewed this issue and confirmed that MCAL is not at risk. MS Defender is returning this as a false positive and we are working to get this reported to Microsoft so it can stop being listed as an issue.

**Reply by pduncliffe** [reply]
I've received a response from Nextgen (Sales) regards my query with them. They've asked MS to correct the false positive. So I would say as soon as MS respond, issue closed.

**Reply by pduncliffe** [reply]
And thanks @JackieK5

**Reply by DanEdomHodge** [reply]
Thank you @pduncliffe @JackieK5 @jonbartels
For reference this was a clean install of 1.4 of the launcher. I believe that MS Defender is flagging that 1.4<4.4.1 for "mirth connect", it's not differentiating between the launcher and server as it scans the registry uninstall settings.
